plugins/filter/json_patch.py

```diff
@@ -0,0 +1,113 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) Stanislav Meduna (@numo68)
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, annotations, division, print_function
+from json import loads
+from typing import TYPE_CHECKING
+from ansible.errors import AnsibleError, AnsibleFilterError, AnsibleOptionsError
+from ansible.module_utils.common.text.converters import to_native
+
+__metaclass__ = type  # pylint: disable=C0103
+
+if TYPE_CHECKING:
+    from typing import Callable
+
+try:
+    import jsonpatch
+
+    HAS_LIB = True
+except ImportError:
+    HAS_LIB = False
+
+
+class FilterModule:
+    """Filter plugin."""
+
+    def json_patch(
+        self, inp: object, op: str, path: str, value: object = None, **kwargs: dict
+    ) -> object:
+
+        if not HAS_LIB:
+            raise AnsibleError(
+                'You need to install "jsonpatch" package prior to running "json_patch" filter'
+            )
+
+        args = {"op": op, "path": path}
```

Comment:
Please note that you have to validate the inputs, i.e. make sure the strings are actually strings, `op` is one of the required strings, and so on.

Reply:
> Please note that you have to validate the inputs, i.e. make sure the strings are actually strings, `op` is one of the required strings, and so on.

How deep should that validation go? For example the `json_patch_recipe` accepts an array of such operations, do I have to check all? The underlying `jsonpatch` library will check and bail out if unacceptable, so I don't think the second level is necessary, but I don't know what the common way of handling in ansible is.

Reply:
I think a general guideline is to at least validate what you are describing in the plugin's documentation. Whether you want to validate more depends on how nice the error messages should be (which depends on how user-friendly the errors returned from the library are - if they simply say "bad input" doing more validation is a good idea, or if they use other parameter names than the ones supplied to the plugin).